syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto";
option java_package = "com.bloock.sdk.bridge.proto";

import "authenticity_entities.proto";
import "record_entities.proto";
import "shared.proto";
import "config.proto";

service AuthenticityService {
    rpc Sign(SignRequest) returns (SignResponse);
    rpc Verify(VerifyRequest) returns (VerifyResponse);
    
    rpc GetSignatures(GetSignaturesRequest) returns (GetSignaturesResponse);
    rpc GetSignatureCommonName(SignatureCommonNameRequest) returns (SignatureCommonNameResponse);
}

message SignRequest {
    ConfigData config_data = 1;
    Record record = 2;
    Signer signer = 3;
}

message SignResponse {
    Signature signature = 1;
    optional Error error = 2;
}

message VerifyRequest {
    ConfigData config_data = 1;
    Record record = 2;
}

message VerifyResponse {
    bool valid = 1;
    optional Error error = 2;
}

message SignatureCommonNameRequest {
    ConfigData config_data = 1;
    Signature signature = 2;
    string hash = 3;
}

message SignatureCommonNameResponse {
    string common_name = 1;
    optional Error error = 2;
}

message GetSignaturesRequest {
    ConfigData config_data = 1;
    Record record = 2;
}

message GetSignaturesResponse {
    repeated Signature signatures = 1;
    optional Error error = 2;
}